src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -271,8 +271,10 @@ async def run_script(job, env, hook, script_name):
     if not hook:
         return
 
-    if not hook.startswith("#!"):
-        hook = f"#!/bin/bash\n{hook}"
+    if hook.startswith("#!"):
+        shebang = shlex.split(hook.splitlines()[0][2:].strip())
+    else:
+        shebang = ["/bin/bash"]
 
     fd, name = tempfile.mkstemp()
```

Comment:
I also asked that you fix up the entirety of this section by:
1. using the temfile context manager
2. and preventing all of this blocking I/O from being executed in a coroutine

Reply:
your comment doesn't matter. Python's builtin `open` and `os.chmod` don't release the GIL so it doesn't matter what the backing filesystem is.

Reply:
@yocalebo it does, ramfs operations will never block since there is nothing to block on. We take advantage of this all over the codebase.

Reply:
@themylogin just for clarity. The backing filesystem doesn't matter. It matters whether or not python's builtin primitives release the GIL for the I/O based calls. To my surprise, `os.chown` `os.open` and at least most of the file I/O based primitives actually do release the GIL.....that's peculiar to me.